### PR TITLE
Lab2- Evgeny Afanasyev

### PR DESCRIPTION
**Pull request by afonyaa**

### CODE REVIEW

File: LAB1.md [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# 1
```
Comment (AKurmazov) [inline LAB1.md]: You may get rid of this one :)

File: app_python/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,14 @@
+# syntax=docker/dockerfile:1
+
+FROM python:3.8-slim-buster
+
+WORKDIR /app_python
+
+COPY requirements.txt requirements.txt
+RUN pip3 install -r requirements.txt
+
+COPY . .
```
Comment (AKurmazov) [inline app_python/Dockerfile:10]: Copying everything into the container is no good, you would probably need only the project sources, and its dependencies

File: app_python/Dockerfile [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+ENTRYPOINT ["python"]
+CMD ["server.py"]
+#CMD ["python3", "-m", "flask", "run", "--host=0.0.0.0"]
```
Comment (AKurmazov) [inline app_python/Dockerfile]:
```suggestion
```

File: app_python/PYTHON.md [hunk 4]
```diff
@@ -0,0 +1,20 @@
...
+## Best practices
+- I did not meet with any testing libraries for python, but due to the reason that it is not static-typed language,
+so I suppose that there are some tools and practices which helps to control the typing in python's programs
+- It is recommended to test any component and interface of system
+- Keeping list of dependencies in `requirements.txt file`
+- Use gitignore
+- Use yaml file for configurations
+
+## Why Python
+- I don't know, I hate it :/
```
Comment (AKurmazov) [inline app_python/PYTHON.md:18]: 😮

File: app_python/README.md [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+## How to run
+- The source file is called `server.py`, located in `app_python` folder
+- To run server, put `python server.py` in command line locating in `app_python` directory
+
+## Run with docker
+- Create image: `docker build -t app_python:latest`
+- Run container: `docker run -d -p 8080:8080 app_python`
+
+## What this app does
+- The application runs server at your localhost on 8000 port and shows current time in Moscow in index page
```
Comment (AKurmazov) [inline app_python/README.md]: The port is 8000 despite the fact that 2 lines above it is written that the mapping is 8080:8080

File: app_python/Dockerfile (hunk 2 above)
Comment (none-word) [reply app_python/Dockerfile:10]: I am agreed with Alexander

File: app_python/PYTHON.md [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+
+## Best practices
+- I did not meet with any testing libraries for python, but due to the reason that it is not static-typed language,
+so I suppose that there are some tools and practices which helps to control the typing in python's programs
+- It is recommended to test any component and interface of system
+- Keeping list of dependencies in `requirements.txt file`
+- Use gitignore
+- Use yaml file for configurations
+
+## Why Python
```
Comment (Maxkoz777) [inline app_python/PYTHON.md:17]: Was there such a point in assignment?)

File: app_python/utils/main.py [hunk 7]
```diff
@@ -0,0 +1,11 @@
+from datetime import datetime
+import pytz
+import yaml
+
+
+def get_current_time_formatted():
+    with open("config.yml", "r", encoding="utf-8") as f:
+        config = yaml.load(f, Loader=yaml.FullLoader)
+    TZ = config["flask"]["timezone"]
```
Comment (Maxkoz777) [inline app_python/utils/main.py:9]: variable naming conventions are violated, there is a plenty of proper ways to rename "TZ" variable

File: app_nodejs/README.md [hunk 8]
```diff
@@ -0,0 +1,7 @@
+## How to run
+- Node js LTS version is required
+- Run `node server.js` will boot server, and user can see the current time in Moscow on browser screen
+
+## Run by docker
+- write ` docker build -t node-app:latest .` in terminal to build image
+- run container by command `ocker run -it -p 8000:8000 node-app`
```
Comment (Homa3030) [inline app_nodejs/README.md]: typo in command 

```suggestion
- run container by command `docker run -it -p 8000:8000 node-app`
```

File: app_python/Dockerfile (hunk 2 above)
Comment (Homa3030) [reply app_python/Dockerfile:10]: You can create .dockerignore file for that or copy only specified files into the container

**Comment by none-word** [comment]
Well done!

**Pull request closed without merging**